datafusion/physical-expr/src/partitioning.rs

```diff
@@ -15,14 +15,43 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! [`Partitioning`] and [`Distribution`] for physical expressions
+//! [`Partitioning`] and [`Distribution`] for `ExecutionPlans`
 
 use std::fmt;
 use std::sync::Arc;
 
 use crate::{expr_list_eq_strict_order, EquivalenceProperties, PhysicalExpr};
 
-/// Partitioning schemes supported by operators.
+/// Partitioning schemes supported by [`ExecutionPlan`]s.
+///
+/// A partition represents an independent stream that an `ExecutionPlan` can
+/// produce in parallel. Each `ExecutionPlan` must produce at least one
+/// partition, and the number of partitions varies based on the input and the
+/// operation performed.
+///
+/// ```text
+///     ▲      ▲      ▲
+///     │      │      │
+///     │      │      │        An ExecutionPlan with 3
+///     │      │      │        output partitions will
+/// ┌───┴──────┴──────┴──┐     produce 3 streams of
+/// │   ExecutionPlan    │     RecordBatches that run in
+/// └────────────────────┘     parallel.
```

Comment:
would be nice to show the input partition as well

Reply:
good idea -- I will do